services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/drs/DrsManagerImpl.java

```diff
@@ -175,4 +177,24 @@ private void validateIdHasVersion(final IdAndVersion idAndVersion) {
             throw new IllegalArgumentException("Object id should include version. e.g syn123.1");
         }
     }
+
+    @Override
+    public AccessUrl getAccessUrl(final Long userId, final String drsObjectId, final String accessId)
+            throws NotFoundException, DatastoreException, UnauthorizedException, IllegalArgumentException {
+        final UserInfo userInfo = this.userManager.getUserInfo(userId);
+        final AccessId accessIdObject = AccessId.decode(accessId);
+        final IdAndVersion objectId = IdAndVersion.parse(drsObjectId);
+        validateAccessIdHasObjectId(objectId, accessIdObject.getSynapseIdWithVersion());
+        final FileHandleUrlRequest urlRequest = new FileHandleUrlRequest(userInfo, accessIdObject.getFileHandleId());
+        final String url = this.fileHandleManager.getRedirectURLForFileHandle(urlRequest);
+        final AccessUrl accessURL = new AccessUrl();
+        accessURL.setUrl(url);
+        return accessURL;
+    }
+
+    private void validateAccessIdHasObjectId(final IdAndVersion objectIdInRequest, final IdAndVersion objectIdASPartOfAccessId) {
+        if (!objectIdASPartOfAccessId.equals(objectIdInRequest)) {
+            throw new IllegalArgumentException("AccessId contains different drsObject Id.");
+        }
+    }
```

Comment:
Let's add a test to DrsManagerImplAutowiredTest that validates an AuthorizedException is thrown when another user ties to download the file.